flow_action_components/CollectionProcessors/force-app/main/default/classes/GetChildCollection.cls

```diff
@@ -19,89 +18,118 @@ global with sharing class GetChildCollection {
+                        throw new InvocableActionException(
+                            'You need to pass either a record or a recordId into this action, representing the entity you want to start with'
+                        );
+                    }
+                    parentSobjType = inputRecord.getSObjectType();
+                    inputRecordId = inputRecord.Id;
                 } else {
-                    recordType = inputRecord.getSObjectType();    
+                    // If we received both a record ID and a record
+                    if (inputRecord != null) {
+                        throw new InvocableActionException(
+                            'You need to pass either a record or a recordId into this action, but you can not pass both'
+                        );
+                    }
+                    parentSobjType = Id.valueOf(inputRecordId).getSObjectType();
                 }
-                System.debug('made it through validation. inputRecord is: ' + inputRecord);
-                for(ChildRelationship curChildRel : recordType.getDescribe().getChildRelationships()){
-                    Schema.DescribeSObjectResult childRelDescribe = curChildRel.getChildSObject().getDescribe();
-                    System.debug('childRelDescribe.getName():::'+childRelDescribe.getName());
-                    if (childRelDescribe.getName() == curRequest.childRelationshipName) {
-                        lookupField = curChildRel.getField();
+                System.debug(
+                    'Made it through validation; parentSobjType is ' +
+                        parentSobjType +
+                        '; inputRecordId is ' +
+                        inputRecordId
+                );
+                for (
+                    ChildRelationship curChildRel : parentSobjType.getDescribe()
+                        .getChildRelationships()
+                ) {
+                    String curChildRelName = curChildRel.getRelationshipName();
+                    System.debug('curChildRelName: ' + curChildRelName);
+                    String curChildObjName = curChildRel.getChildSObject()
+                        .getDescribe()
+                        .getName();
+                    System.debug('curChildObjName: ' + curChildObjName);
+
+                    // If the provided childRelationshipName matches either the
+                    // API name of the child relationship OR the API name of the
+                    // child SObject, use this relationship
+                    if (
+                        curRequest.childRelationshipName == curChildRelName ||
+                        curRequest.childRelationshipName == curChildObjName
+                    ) {
+                        lookupFieldName = curChildRel.getField()
+                            .getDescribe()
+                            .getName();
+                        childSobjTypeName = curChildObjName;
+                        break;
```

Comment:
Adding `break` statement here for performance reasons. No reason to keep looping once a match is found.